server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageFullViewFactory.java

```diff
@@ -98,11 +114,47 @@ public MessageFullView fromMetaDataWithContent(MetaDataWithContent message) thro
                 .date(getDateFromHeaderOrInternalDateOtherwise(mimeMessage, message))
                 .textBody(textBody)
                 .htmlBody(htmlBody)
-                .preview(preview)
+                .preview(messageProjection.getPreview())
                 .attachments(getAttachments(message.getAttachments()))
                 .build();
     }
 
+    private MessageFastViewPrecomputedProperties retrieveProjection(MessageContent messageContent,
+                                                                    MessageId messageId, Supplier<Boolean> hasAttachments) {
+        return Mono.from(fastViewProjection.retrieve(messageId))
+            .onErrorResume(throwable -> fallBackToCompute(messageContent, hasAttachments, throwable))
```

Comment:
Should we try to store the computed preview in case of a fallback ?

Reply:
That is done in another pr.

Reply:
error when retrieving doesn't mean 100% the data was not there. But most probably the cassandra is not doing its job and try to store probably doesn't effect.